### PR TITLE
[incubator/elasticsearch] Detect allocation without using JQ

### PR DESCRIPTION
**Pull request by KongZ**
This PR removes the JQ dependency from `pre-stop-hook.sh`
It will detect allocation from `_cat/shards` API. If shutting down node name does not appear in shards allocation, then exit the script.

### CODE REVIEW

**Comment by yanns** [comment]
@KongZ Thank for you PR, I could use it.
One thing I added is a post start hook to enable the node again in the logical ES cluster:
```
  post-start-hook.sh: |-
    #!/bin/bash
    set -e
    NODE_NAME=${HOSTNAME}
    CLUSTER_SETTINGS=$(curl -s -XGET "http://${DISCOVERY_SERVICE}:9200/_cluster/settings")
    if echo "${CLUSTER_SETTINGS}" | grep -E "${NODE_NAME}"; then
      echo "Activate node ${NODE_NAME}"
      curl -s -XPUT -H 'Content-Type: application/json' "http://${DISCOVERY_SERVICE}:9200/_cluster/settings" -d "{
        \"transient\" :{
            \"cluster.routing.allocation.exclude._name\" : null
        }
      }"
    fi
    echo "Node ${NODE_NAME} is ready to be used"
    # any other exit code than 0 is considered as an error by Kubernetes
    exit 0
```

**Comment by unguiculus** [comment]
@KongZ What's the status? Should @yanns change be incorporated into this PR?

**Comment by yanns** [comment]
I remove the ` set -e` to be able to start a new cluster from scratch.
Last version:
```
  post-start-hook.sh: |-
    #!/bin/bash
    # we don't want to stop the deployment if this script if failing
    # therefore: no 'set -e'

    NODE_NAME=${HOSTNAME}

    CLUSTER_SETTINGS=$(curl -s -XGET "http://${DISCOVERY_SERVICE}:9200/_cluster/settings")
    if echo "${CLUSTER_SETTINGS}" | grep -E "${NODE_NAME}"; then
      echo "Activate node ${NODE_NAME}"
      curl -s -XPUT -H 'Content-Type: application/json' "http://${DISCOVERY_SERVICE}:9200/_cluster/settings" -d "{
        \"transient\" :{
            \"cluster.routing.allocation.exclude._name\" : null
        }
      }"
    fi

    echo "Node ${NODE_NAME} is ready to be used"
    # any other exit code than 0 is considered as an error by Kubernetes
    exit 0
```

It works very well. We upgraded the K8S cluster and the ES cluster several times and could maintain the ES cluster green during the complete deployment.

**Comment by KongZ** [comment]
@unguiculus merged `post-start-hook.sh` to this PR
@yanns thanks

**Comment by KongZ** [comment]
For anyone who wish to try full Elasticsearch Chart, you can checkout my personal repository here
https://github.com/KongZ/elasticsearch-chart

I have use this chart in my production for several months.

I will contribute full chart into this repository but it gonna be long procedures. The chart is changing almost entire templates.

File: incubator/elasticsearch/templates/configmap.yaml [hunk 1]
```diff
@@ -93,60 +93,35 @@ data:
 {{- end }}
   pre-stop-hook.sh: |-
     #!/bin/bash
-    set -e
```
Comment (yanns) [inline incubator/elasticsearch/templates/configmap.yaml:120]: Sorry if I was not clear.
I kept the `set -e` for the `pre-stop-hook` (I removed from the `post-stop-hook`)

Comment (KongZ) [reply incubator/elasticsearch/templates/configmap.yaml:120]: @yanns  `set -e` does not matter on `pre-stop-hook.sh` too. So I remove it and make a code as short as possible.

**Comment by mechinn** [comment]
@KongZ I think you accidentally missed the change to add post-start-hook.sh to data-statefulset.yaml

**Comment by KongZ** [comment]
@mechinn thanks. fixed it

**Pull request merged**